Assets/Stack mini game/Scripts/StackMinigameManager.cs

```diff
@@ -0,0 +1,158 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class StackMinigameManager : MonoBehaviour

```

Comment:
Some of this stuff I guess you could put in a separate class, like ```CubeCutter``` or something. 

But having a big GameManager-style class is pretty standard in Unity lol. I would try to make the functions reasonably small though.